subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c

```diff
@@ -247,15 +247,30 @@ static void sync_rest_client_data(struct nrf_cloud_rest_context *const rest_ctx,
 
 static int do_rest_client_request(struct nrf_cloud_rest_context *const rest_ctx,
 	struct rest_client_req_context *const req,
-	struct rest_client_resp_context *const resp)
+	struct rest_client_resp_context *const resp,
+	bool check_status_good, bool expect_body)
 {
 	int ret = rest_client_request(req, resp);
 
 	sync_rest_client_data(rest_ctx, req, resp);
 
-	return ret;
-}
+	if (ret) {
+		LOG_DBG("REST client request failed with error code %d", ret);
+		return ret;
+	} else if (rest_ctx->status == NRF_CLOUD_HTTP_STATUS_NONE) {
+		LOG_DBG("REST request endpoint closed connection without reply.");
+		return -ESHUTDOWN;
+	} else if (check_status_good && rest_ctx->status != NRF_CLOUD_HTTP_STATUS_OK &&
```

Comment:
IIRC, the Nordic coding style requires parentheses in conditional expressions to make the order of operations explicit.